src/shogun/multiclass/tree/CARTree.cpp

```diff
@@ -281,10 +281,30 @@ bool CARTree::train_machine(std::shared_ptr<Features> data)
 	{
 		prune_by_cross_validation(dense_features,m_folds);
 	}
-
 	return true;
 }
 
+void CARTree::compute_feature_importance(std::shared_ptr<bnode_t> node)
+{
+	auto left = node->left();
+	auto right = node->right();
+	if (left && right)
+	{
+		m_feature_importances[node->data.attribute_id] +=
+		    (node->data.impurity * node->data.total_weight -
+		     left->data.impurity * left->data.total_weight -
+		     right->data.impurity * right->data.total_weight);
+	}
+	if (left->data.num_leaves > 1)
```

Comment:
just a thought no need to implement this here right now, but instead of this recursive call we could fill up an array and then calculate the importance in a loop... that later could be openmp-ed? :)